client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/pattern-selector.tsx

```diff
@@ -57,42 +27,30 @@ const PatternSelector = ( {
 			</div>
 			<div className="pattern-selector__body">
 				<div className="pattern-selector__block-list" role="listbox">
-					{ patterns?.map( ( item: Pattern ) => (
-						<div
-							key={ item.id }
-							aria-label={ item.name }
-							tabIndex={ 0 }
-							role="option"
-							aria-selected={ isSelected( item.id ) }
-							className={ classNames( { '--pattern-selected': isSelected( item.id ) } ) }
-							onClick={ () => handleSelectedPattern( item ) }
-							onKeyUp={ handleKeyboard( () => setSelectedPattern( item ) ) }
+					{ patterns?.map( ( item: Pattern, index: number ) => (
+						<PatternPreviewAutoHeight
+							key={ `${ index }-${ item.id }` }
+							url={ getPatternPreviewUrl( item.id, locale ) }
+							patternId={ item.id }
 						>
-							<iframe
-								title={ item.name }
-								src={ getPatternPreviewUrl( item.id ) }
-								frameBorder="0"
-								aria-hidden
-								tabIndex={ -1 }
-							></iframe>
-						</div>
+							<div
+								aria-label={ item.name }
+								tabIndex={ 0 }
+								role="option"
+								aria-selected={ false }
+								onClick={ () => onSelect( item ) }
+								onKeyUp={ handleKeyboard( () => onSelect( item ) ) }
+							>
+								<iframe title={ item.name } frameBorder="0" aria-hidden tabIndex={ -1 }></iframe>
```

Comment:
Is there any reason we need to keep the iframe here? I thought we could render the iframe inside `PatternPreviewAutoHeight` and then we don't need to clone element and assume the children of the children is an iframe. What do you think?

For example,
```js
<div>
  <PatternPreviewFrame ... />
</div>
```